common/serial/port_io.h

```diff
@@ -0,0 +1,43 @@
+#ifndef PORT_IO__H
+#define PORT_IO__H
+
+#include <arch.h>
+
+class port_io {
+
+public:
+	port_io(arch_handler *arch) : m_arch(arch) 
+	{
+
+	}
+
+	~port_io(void) 
+	{
+		delete m_arch;
+	}
+
+	void port_write(uint16_t port, uint8_t value)
+	{
+		m_arch->port_write(port, value);
```

Comment:
While I generally agree with the NASM approach, inline may be more appropriate in this scenario (one liners if done inline, probably no less than 10-20 if done separately.)